src/libpriv/rpmostree-refts.cxx

```diff
@@ -76,31 +77,180 @@ rpmostree_refts_unref (RpmOstreeRefTs *rts)
 
 namespace rpmostreecxx
 {
+static inline std::pair<std::string_view, std::string_view>
+split_filepath (std::string_view path)
```

Comment:
A while ago we started compiling in C++ mode with the *primary* goal of being able to use https://cxx.rs/ as a bridge between Rust and C (which has kind of worked and kind of made things worse in other ways).

I'll be honest, I only know enough C++ to be dangerous and trying to review C++ string code is definitely in the "dangerous" territory.

Previously this code was just a thin shim to librpm, but it's now growing nontrivial logic and the main idea was that new code very specifically like this would be in Rust.

I'm not going to say we should block on this to be clear.

(It's not just memory safety, it's also that functions like this really cry out for unit testing, and that's just IMO very nice and convenient to do in Rust, and much less so in C++.  But, part of that is just us not investing in it of course)

Reply:
Honestly, I wrote the logic in C++ since that's where the existing code was. I'm more than happy to move more of the logic into the Rust side if that's preferred. Let me take a look this weekend to see if I can rework it a bit.